overlord/snapstate/snapstate.go

```diff
@@ -1892,9 +1892,7 @@ func Remove(st *state.State, name string, revision snap.Revision, flags *RemoveF
 		removeHook := SetupRemoveHook(st, snapsup.InstanceName())
 		addNext(state.NewTaskSet(removeHook))
 		prev = removeHook
-	}
 
-	if removeAll {
```

Comment:
for next time: this is a drive-by which is ok,  we tend to put "drive-by" in the commit message for those, also if possible to do them first in the commit set

Reply:
okay, I will try to have the commit mention drive-by and put this commit first in the PR